returnn/datasets/hdf.py

```diff
@@ -113,7 +113,6 @@ def add_file(self, filename):
       prev_target_keys = self.target_keys
     if 'targets' in fin:
       self.target_keys = sorted(
-        set(fin['targets/labels'].keys()) |
```

Comment:
Why is this change relevant?

Reply:
Yep, I also wonder. When reading the title of this PR, my first thought was, why should this be in any way specific about `HDFDataset`?

Reply:
Yep, title is probably not optimal, but it is what connects the two commits 😅 Could have also opened two PRs. Commit messages are clear, I hope.

Simplest example:
You have a trained net, one input, one sparse output, i.e. with vocabulary/labels. To do search you need inputs and output labels. So a _minimal_ HDF would contain the fields: `inputs`, `seqLengths` and `targets/labels/classes`. It does not contain `targets/data/classes`.
Now the question is whether `self.target_keys` refers to any keys the dataset knows of or whether it refers to the data  provided by the dataset. It has to be the latter one. For example the second dim of `seqLengths` is `len(self.target_keys)` and we only have sequence lengths for `data`, not for `classes`.

Reply:
You should add some comment to clarify what `target_keys` is used for. So you say it has to be the latter one because of `seqLengths`. But maybe this is used inconsistently also for other things where the first case makes more sense? Anyway, add a comment about `seqLengths`, and maybe some more.

Reply:
I think this comment I have now is enough. `self.labels` is independent from anything else, it is only used in `serialize_data()`. So including its keys in `self.target_keys` was kind of wrong in the first place. If anything, it was an assertion that the keys match, but I want to relax that.